src/libsyntax/attr.rs

```diff
@@ -299,16 +299,16 @@ pub fn find_crate_name(attrs: &[Attribute]) -> Option<InternedString> {
 }
 
 /// Find the value of #[export_name=*] attribute and check its validity.
-pub fn find_export_name_attr(diag: &Handler, attrs: &[Attribute]) -> Option<InternedString> {
+pub fn find_export_name_attr(diag: Option<&Handler>, attrs: &[Attribute]) -> Option<InternedString> {
```

Comment:
I think both those contexts have access to a `ty::ctxt` which I believe should have a handle to this via some methods (probably through the `Session` itself)?